cmd/skopeo/copy.go

```diff
@@ -55,12 +82,20 @@ func copyHandler(context *cli.Context) {
 	}
 	signBy := context.String("sign-by")
 
-	manifest, _, err := src.GetManifest([]string{utils.DockerV2Schema1MIMEType})
+	mimeTypes := []string{manifest.DockerV2Schema2MIMEType}
+	// TODO(runcom, mitr): will refactor in followup where copy becomes a library
+	// if not OCI image destination, then ask for v2s1 manifest as well for config
+	// this is ugly I know, we'll refactor
+	if !strings.HasPrefix(context.Args()[1], ociPrefix) {
+		mimeTypes = append(mimeTypes, manifest.DockerV2Schema1MIMEType)
+	}
```

Comment:
This still needs solving.

From the POV of `copy.go`, it seems it would be best to:
- Always ask for OCI?, v2s2, v2s1, in that order (we might want to parametrize this depending on destination later, but now it seems unnecessary)
- Implement `manifestBlobs`/`Image.LayerDigests` for OCI and v2s2 as well. Then `copy.go` won’t care what the source uses.
- Keep the version check in `OCIDestination` to refuse v2s1.

Or do we need to keep preferring v2s1 because of OpenShift or something? AFAICS `skopeo copy` is a new functionality anyway, so does any of the other projects depend on skopeo preferring v2s1 right now?